packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1472,45 +1474,54 @@ abstract class ParentDataWidget<T extends RenderObjectWidget> extends ProxyWidge
   @override
   ParentDataElement<T> createElement() => ParentDataElement<T>(this);
 
-  /// Subclasses should override this method to return true if the given
-  /// ancestor is a RenderObjectWidget that wraps a RenderObject that can handle
-  /// the kind of ParentData widget that the ParentDataWidget subclass handles.
-  ///
-  /// The default implementation uses the type argument.
-  bool debugIsValidAncestor(RenderObjectWidget ancestor) {
+  /// Checks if this widget can apply its parent data to the provided
+  /// `renderObject`.
+  bool debugIsValidRenderObject(RenderObject renderObject) {
```

Comment:
docs should probably mention debugTypicalAncestorWidget and also mention when this is called in the widget lifecycle

Reply:
Done.